src/Corax/Indexing/IndexWriter.IndexEntryBuilder.cs

```diff
@@ -37,15 +39,42 @@ public void Boost(float boost)
             _parent.BoostEntry(_entryId, boost);
         }
 
-        public void Init(long entryId, int termsPerEntryIndex)
+        public void Init(long entryId, int termsPerEntryIndex, Slice documentId)
         {
             Active = true;
+            _mapFinishedSuccessfully = false;
             _entryId = entryId;
             _termPerEntryIndex = termsPerEntryIndex;
+            _documentId = documentId;
         }
 
+        /// <summary>
+        /// This method must be called before dispose to indicate there were no exceptions during indexing.
+        /// If this is not called, we will start the rollback procedure (see more at Dispose()).
+        /// This allows us to avoid using a try...catch...finally block, thereby avoiding the overhead associated with the catch block.
```

Comment:
Is there a significant cost to `catch` beyond `finally`? After all `using` is the same

Reply:
Not exactly, catch clause generate a lot of code:

https://gist.github.com/maciejaszyk/9443d140897b244487027da6115a6b42
```
| Method             | Mean      | Error     | StdDev    | Median    |
|------------------- |----------:|----------:|----------:|----------:|
| BenchmarkCatchFree |  7.024 ms | 0.1367 ms | 0.1825 ms |  6.894 ms |
| BenchmarkWithCatch | 15.965 ms | 0.1021 ms | 0.0797 ms | 15.924 ms |
```

See ASM here:
https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQ64GYAEicDKA9lAKbAAWAlgHYDmOIOAkgCJUQAOREYARgBtS6AN7ocEvLmJlKtOgAoAlOMli0kzTlUSAvju0bJ2PABYcrIsoMj9RiQZNxz7Lj1LX7OW+juZc+ADC3o4B5gCyyiFemjLk1PQ4AB44ALw4NACuAgIA3AaawABOAJ7qWhUSKek0pADuhCTx8sr5MZVJAHSWrQWSdpU4AGa0YDllfRVJAPzdHNwQHkptgwP9BpNSZjgAqlHllXAADI2yCQzVGfW97VXdVssGaw5eTuaBACb7m3FyiZdZHIrSrFCa3KZpK4NX7nG6DTRdHqPcHPLQAYzAwDRFAOFVRmhGNDGAjB8MkMzmbkWcMqzzsuiAA==

Reply:
I am pleasantly surprised how good the `try..catch` assembler is now.